lib/go/thrift/http_transport.go

```diff
@@ -40,14 +41,23 @@ func NewThriftHandlerFunc(processor TProcessor,
 
 // gz transparently compresses the HTTP response if the client supports it.
 func gz(handler http.HandlerFunc) http.HandlerFunc {
+	sp := &sync.Pool{
+		New: func() interface{} {
+			return gzip.NewWriter(nil)
+		}}
```

Comment:
nit: style wise I'd slightly prefer `},\n}`, but we are not very strictly on style things anyways so I'm not feeling strongly.

Reply:
No worries, will change this before merging.